arithmetic_analysis/newton_raphson.py

```diff
@@ -4,34 +4,41 @@
 # quickly find a good approximation for the root of a real-valued function
 from __future__ import annotations
 
-from decimal import Decimal
-from math import *  # noqa: F403
+from sympy import diff, symbols, sympify
 
-from sympy import diff
 
-
-def newton_raphson(
-    func: str, a: float | Decimal, precision: float = 10**-10
-) -> float:
+def newton_raphson(func: str, a: float, precision: float = 10**-10) -> float:
     """Finds root from the point 'a' onwards by Newton-Raphson method
     >>> newton_raphson("sin(x)", 2)
     3.1415926536808043
     >>> newton_raphson("x**2 - 5*x +2", 0.4)
-    0.4384471871911695
+    0.4384471871911696
     >>> newton_raphson("x**2 - 5", 0.1)
     2.23606797749979
     >>> newton_raphson("log(x)- 1", 2)
     2.718281828458938
     """
```

Comment:
Please add a test with 0 precision and -1 precision.
Tests should also validate that this algorithm generates the same results as `scipy.optimize.newton()`.

Reply:
a negative precision value doesn't make sense in the context of the Newton-Raphson method since it implies an infinite level of accuracy....

Reply:
# SO RAISE A ValueError EXCEPTION.

Reply:
```suggestion
    >>> from scipy.optimize import newton
    >>> all(newton_raphson("log(x)- 1", 2) == newton("log(x)- 1", 2) for precision in 10, 100, 1000, 10000))
    True
    >>> newton_raphson("log(x)- 1", 2, 0)
    Traceback (most recent call last):
        ...
    ValueError: precision must be greater than zero
    >>> newton_raphson("log(x)- 1", 2, -1)
    Traceback (most recent call last):
        ...
    ValueError: precision must be greater than zero
    """
```

Reply:
How does `scipy.optimize.newton()` deal with a precision of 0 and -1?